src/interceptors/add_client_secret_to_params.js

```diff
@@ -0,0 +1,12 @@
+/**
+   Read `clientSecret` from `ctx` and add it to `params`
+
+   Changes to `ctx`:
+
+   - add `params.client_secret`
+ */
+export default class AddClientSecretToParams {
+  enter({ clientSecret, params, ...ctx }) {
+    return { ...ctx, clientSecret, params: { ...params, client_secret: clientSecret } };
```

Comment:
If client secret is not set, this could throw an error saying that client secret is missing. I hit this couple of times in my tests yesterday and was wondering why things don't work. The downside then is that client secret is made mandatory, which it today is yes, but maybe not for all token exchanges in the future?

Reply:
I added the check.

If in the future we add token exchanges that don't require a client secret, I think we can make the check based on the params given to this interceptor.